### PR TITLE
chore(api): proto definition of passkeys endpoints (user service)

### PR DESCRIPTION
**Pull request by stebenz**
### Definition of Ready

- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

Defintions:

registration -> https://webauthn.guide/#registration
authentication(verification) -> https://webauthn.guide/#authentication

### CODE REVIEW

**Comment by muhlemmer** [comment]
Related: #5469

File: proto/zitadel/user/v2alpha/auth.proto [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    (validate.rules).string = {min_len: 1, max_len: 200},
+    (google.api.field_behavior) = REQUIRED,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      min_length: 1;
+      max_length: 200;
+      example: "\"https://example.com/passkey/register?userID={{.UserID}}&code={{.Code}}&orgID={{.OrgID}}\"";
+      description: "\"Optionally set a url_template, which will be used in the mail sent by ZITADEL to guide the user to your registration page. If no template is set, the default ZITADEL url will be used.\""
+    }
+  ];
+}
```
Comment (muhlemmer) [inline proto/zitadel/user/v2alpha/auth.proto:28]: `optional` field contradicts `REQUIRED` valdiation option.

File: proto/zitadel/user/v2alpha/user_service.proto [hunk 2]
```diff
@@ -254,3 +296,105 @@ message VerifyEmailRequest{
...
+  // if no medium is specified, an email is sent with the default url
+  oneof medium {
+    SendPasskeyRegistrationLink send_link = 2;
+    ReturnPasskeyRegistrationCode return_code = 3;
+  }
+}
+
+message CreatePasskeyRegistrationLinkResponse{
+  zitadel.object.v2alpha.Details details = 1;
+  string code = 2;
```
Comment (muhlemmer) [inline proto/zitadel/user/v2alpha/user_service.proto:399]: code should be `optional` as it is only returned if the `ReturnPasskeyRegistrationCode` is present in the request.
See `verification_code` in `SetEmailResponse`:

https://github.com/zitadel/zitadel/blob/483f88f0cae4e3ab940e63629913555ef887346a/proto/zitadel/user/v2alpha/user_service.proto#L268-L272

File: proto/zitadel/user/v2alpha/auth.proto (hunk 1 above)
Comment (livio-a) [reply proto/zitadel/user/v2alpha/auth.proto:28]: yes that doesn't make sense 🤦

Comment (livio-a) [reply proto/zitadel/user/v2alpha/auth.proto:28]: i've fixed it (and on other messages)

File: proto/zitadel/user/v2alpha/user_service.proto (hunk 2 above)
Comment (livio-a) [reply proto/zitadel/user/v2alpha/user_service.proto:399]: does it make sense to change it on the request as well as they're only needed if the calling user is permitted directly?

**Review by livio-a — APPROVED** [review]

**Pull request merged**